### ISSUE TITLE
Can't add a Python tool that uses `from typing import List` and `def ip_info(ips: List[str]) -> dict:`

### ISSUE THREAD

**Issue by thomassuedbroecker** [i-am-bee/bee-ui#159]
## Description 
It is not possible to add a Python tool that contains.

```python
from typing import List

def ip_info(ips: List[str]) -> dict:
```
I tried a version without a list, and this version was possible to add.

```python
# from typing import List

def ip_info(ips: str) -> dict:
```

## To Reproduce
1. Go to 'Tools'
2. Click on 'new Tool'
3. Insert the following example code for the tool definition:
```python
import json
import requests
from typing import List

def ip_info(ips: List[str]) -> dict:
   """
   Get information about an IP address, such as location, company, and carrier name.
   : param ips: A list of IP addresses in the 255.255.255.255 format
   : return: Information about the IP address
   """
   ips_info = []
   for ip in ips:
      print(f"ip: {ip}")
      response = requests.get (f"https://ipinfo.io/{ip}/geo")
      
      print(f"response: {response}")
      response.raise_for_status()
      
      print(f"response: {response.json()}")
      ips_info.append(response.json())
   
   return json.dumps(ip_infos)
```
4. Press add
5. See error developer tools Bee UI and Bee API

* Bee UI

<img width="1773" alt="2024-12-18_14-28-49" src="https://github.com/user-attachments/assets/521223f0-19f3-4c47-8774-8d82d2efb4b5" />

* Bee API

<img width="467" alt="2024-12-18_14-29-30" src="https://github.com/user-attachments/assets/b509b0c0-1cc5-4648-9f08-019ede5ddff2" />

## Expected behavior
The tool should be added.

## Desktop (please complete the following information):
- OS: macOS]
- Browser: Version 131.0.6778.139 (Official Build) (arm64)
- Version: Bee API, Bee UI, Stack only Infrastructure actual version from today 2024.12.24.
- Container engine: Podman

## Additional context
No additional images.

**Comment by abdullahdevrel** [comment]
I am not familiar with Bee Stack, but I work for IPinfo. Do you have an adblocker running? Try disabling it and see if that works.

**Comment by thomassuedbroecker** [comment]
@abdullahdevrel @Tomas2D 

This problem is related to saving the tool and not using the tool at the runtime. The tool can never be saved, if you have this kind of Python code `from typing import List` in your custom tool.

The same behavior is also for the other custom Python tool I tried; you can see it here https://suedbroecker.net/2024/12/19/create-custom-bee-agent-with-a-custom-python-weather-tool-a-step-by-step-guide/

Do you have an example of it working with such a definition where you could reproduce the problem I wrote in the **To Reproduce**  section?

**Comment by Tomas2D** [comment]
Good catch @thomassuedbroecker. The problem lay in generating JSON Schema based on the function signature.

Fix: https://github.com/i-am-bee/bee-code-interpreter/pull/12

In addition, we should add proper error handling to the UI. A notification would say that the generated code is not valid. Unfortunately, we must propagate such information from the `bee-code-interpreter`.

cc @mmurad2, @geneknit, @JanPokorny

**Comment by JanPokorny** [comment]
@thomassuedbroecker Hello, thanks for bringing this up.

We'll have a fix ready soon.

As a workaround for the meantime, you can use the modern syntax like this:

```
def ip_info(ips: list[str]) -> dict:
```

...or you can use a qualified import:

```
import typing

def ip_info(ips: typing.List[str]) -> dict:
```

**Comment by Tomas2D** [comment]
Fixed in the new bee-code-interpreter version (0.0.28). I just updated the bee stack.

**Comment by thomassuedbroecker** [comment]
Thanks
